src/drivers/intel/cavs/ipc.c

```diff
@@ -23,6 +23,35 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#if CAVS_VERSION >= CAVS_VERSION_1_8
+
+/** \brief Bit set by host to indicate msg payload is in regs */
+#define IPC_DIPCTDR_MSG_REG		BIT(30)
```

Comment:
We have 60 bits to play with, no? Why use only 16+1 only?

Reply:
TDA(ck) is unavailable, it is RO on the target's side. All the bits available in TDD are used for payload, and there could be one more allocated in TDR indeed, but 3 bytes are still enough for the one and only use case we may have due to the limitations created by "bi-directional" use of the registers:

DSP has to respond via the mailbox, since has no RW access to the registers
(except for Doorbell Ack but this is not symmetric with a pair TDR/TE).
Therefore any "short" LP request must be responded with powering on the mailbox
memory first.

DSP does not know when the response is consumed by the host, so the mailbox
memory must be kept on.

Both are OK short term for limited use, planned "exit power gating" flow. More
aggressive power savings would require use of *T and *I register sets by the
request-response flow.